### PR TITLE
Feature/refactor use cases working on vendorconsents

### PR DESCRIPTION
**Pull request by alextremp**
This PR:

* Sets the VendorConsents object in the middle of the domain
* Uses the ConsentString object to restore a VendorConsents from the euconsent cookie

### CODE REVIEW

File: src/cmp/infrastructure/repository/ConsentStringVendorConsentsRepository.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    consentRepository,
+    vendorListRepository
+  }) {
+    this._getStoredConsent = getStoredConsent({consentRepository})
+    this._createVendorConsents = createVendorConsents({vendorConsentsFactory})
+    this._getGlobalVendorList = getGlobalVendorList({vendorListRepository})
+  }
+
+  getVendorConsents({allowedVendorIds} = {}) {
+    return Promise.resolve()
```
Comment (victuxbb) [inline src/cmp/infrastructure/repository/ConsentStringVendorConsentsRepository.js]: Here it's not necessary to start with a Promise.resolve empty, getStoredConsent and getGlobalVendorList are returning Promise already, so there is no possibility to throw an error outside the arrow function or outside Promise flow.

File: src/cmp/infrastructure/repository/CookieConsentRepository.js [hunk 2]
```diff
@@ -3,28 +3,51 @@
...
-    return Promise.resolve().then(() =>
-      this._readCookie({cookieName: VENDOR_CONSENT_COOKIE_NAME})
-    )
-  }
-
-  _readCookie({cookieName}) {
+  getConsent() {
     return Promise.resolve()
-      .then(() => `; ${this._dom.cookie}`.split(`; ${cookieName}=`))
+      .then(() =>
```
Comment (victuxbb) [inline src/cmp/infrastructure/repository/CookieConsentRepository.js]: Same here, there is no need to start with a Promise empty

File: src/cmp/application/GetConsentStatusUseCase.js [hunk 3]
```diff
@@ -9,12 +9,11 @@ export default class GetConsentStatusUseCase {
...
-      .then(() => this._consentRepository.getConsentData())
-      .then(consentData => {
-        return consentData === undefined
-          ? CONSENT_STATUS_NOT_ACCEPTED
-          : CONSENT_STATUS_ACCEPTED
-      })
+      .then(() => this._consentRepository.getConsent())
+      .then(
+        consent =>
+          consent ? CONSENT_STATUS_ACCEPTED : CONSENT_STATUS_NOT_ACCEPTED
```
[the remark this answers is not in the document]
Comment (alextremp) [reply src/cmp/application/GetConsentStatusUseCase.js]: done :)

File: src/cmp/infrastructure/repository/ConsentStringVendorConsentsRepository.js (hunk 1 above)
Comment (alextremp) [reply src/cmp/infrastructure/repository/ConsentStringVendorConsentsRepository.js]: done :)

File: src/cmp/infrastructure/repository/CookieConsentRepository.js (hunk 2 above)
Comment (alextremp) [reply src/cmp/infrastructure/repository/CookieConsentRepository.js]: done :)

**Pull request merged**